controller/src/beerocks/master/tasks/client_steering_task.cpp

```diff
@@ -174,42 +175,51 @@ void client_steering_task::steer_sta()
                          &client_disallow_event);
     }
 
-    // Send disconnect (802.11v)
-    auto bss_steer_request =
-        message_com::create_vs_message<beerocks_message::cACTION_CONTROL_CLIENT_BSS_STEER_REQUEST>(
-            cmdu_tx, id);
-
-    if (bss_steer_request == nullptr) {
-        LOG(ERROR) << "Failed building ACTION_CONTROL_CLIENT_BSS_STEER_REQUEST message!";
+    // Send STEERING request
+    if (!cmdu_tx.create(0, ieee1905_1::eMessageType::CLIENT_STEERING_REQUEST_MESSAGE)) {
+        LOG(ERROR) << "cmdu creation of type CLIENT_STEERING_REQUEST_MESSAGE, has failed";
         return;
     }
-    bss_steer_request->params().mac            = network_utils::mac_from_string(sta_mac);
-    bss_steer_request->params().disassoc_timer = disassoc_timer;
-    bss_steer_request->params().bssid.mac      = network_utils::mac_from_string(hostap_mac);
-    bss_steer_request->params().bssid.channel  = database.get_node_channel(hostap_mac);
-
-    if (disassoc_imminent) {
-        TASK_LOG(DEBUG) << "diassoc_imminent enabled";
-        bss_steer_request->params().disassoc_imminent = 1;
-    } else {
-        TASK_LOG(DEBUG) << "diassoc_imminent disabled";
-        bss_steer_request->params().disassoc_imminent = 0;
+
+    auto steering_request_tlv = cmdu_tx.addClass<wfa_map::tlvSteeringRequest>();
+
+    if (!steering_request_tlv) {
+        LOG(ERROR) << "addClass wfa_map::tlvSteeringRequest failed";
+        return;
     }
 
     auto current_ap_mac = database.get_node_parent(sta_mac);
 
+    steering_request_tlv->request_flags().request_mode =
+        wfa_map::tlvSteeringRequest::REQUEST_IS_A_STEERING_MANDATE_TO_TRIGGER_STEERING;
+    steering_request_tlv->request_flags().btm_disassociation_imminent_bit = !!disassoc_imminent;
+
+    steering_request_tlv->btm_disassociation_timer() = disassoc_timer;
```

Comment:
This is not correct. The disassoc_timer given to this function is in units of 100ms, while the btm_disassociation_timer is in units of 1TU (1024µs).

I think we should convert the uses of the disassociation timer to use TUs everywhere since that's what goes in the BTM request anyway. The bml and cli don't need to be changed. #323